src/applications/edu-benefits/components/SchoolSelectField.jsx

```diff
@@ -0,0 +1,228 @@
+              {'Search Schools'}
+            </button>
+          </div>
+          {showInstitutions && <div>
+            {`${searchResultsCount} results for ${institutionQuery}`}
+            {institutions.map(({ city, facilityCode, name, state }, index) => (
+              <div key={index}>
+                <div className="radio-button">
+                  <input
+                    autoComplete="false"
+                    checked={facilityCodeSelected === facilityCode}
+                    id={`radio-buttons-${index}`}
+                    name={name}
+                    type="radio"
+                    onKeyDown={this.onKeyDown}
+                    onChange={() => this.handleOptionClick({ city, facilityCode, name, state })}
+                    value={facilityCode}/>
+                  <label
+                    id={`institution-${index}-label`}
+                    htmlFor={`radio-buttons-${index}`}>
+                    <span className="institution-name">{name}</span>
+                    <span className="institution-city-state">{`${city}, ${state}`}</span>
+                  </label>
+                </div>
+              </div>))
+            }
+          </div>
+          }
+          {showInstitutionsLoading && <div>
+            <LoadingIndicator message={`Searching ${institutionQuery}...`}/>
+          </div>
+          }
+          {showPaginationLoading && <div>
+            <LoadingIndicator message={`Loading page ${currentPageNumber} results for ${institutionQuery}...`}/>
+          </div>
+          }
+        </div>
+        {showPagination && <Pagination
+          page={currentPageNumber} pages={pagesCount} onPageSelect={this.handlePageSelect}/>
+        }
+      </fieldset>
+    );
+  }
+}
+
+const mapStateToProps = (state, props) => {
+  const currentPageNumber = selectCurrentPageNumber(state);
+  const facilityCodeSelected = props.formData ? props.formData : '';
+  const institutionQuery = selectInstitutionQuery(state);
+  const institutions = selectInstitutions(state);
+  const institutionSelected = selectInstitutionSelected(state);
+  const pagesCount = selectPagesCount(state);
+  const searchInputValue = selectSearchInputValue(state);
+  const searchResultsCount = selectSearchResultsCount(state);
+  const showInstitutions = selectShowInstitutions(state);
+  const showInstitutionsLoading = selectShowInstitutionsLoading(state);
+  const showPagination = selectShowPagination(state);
+  const showPaginationLoading = selectShowPaginationLoading(state);
+
+  return {
+    currentPageNumber,
+    facilityCodeSelected,
+    institutionQuery,
+    institutions,
+    institutionSelected,
+    pagesCount,
+    searchInputValue,
+    searchResultsCount,
+    showInstitutions,
+    showInstitutionsLoading,
+    showPagination,
+    showPaginationLoading,
+  };
+};
+const mapDispatchToProps = {
+  searchInputChange,
+  searchSchools,
+  selectInstitution
+};
+
+SchoolSelectField.PropTypes = {
```

Comment:
Made the view very dumb- most of the logic is in the reducer. Opted to go that route so that it was easy to restore the search state on the review page.